azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/ConfigMojo.java

```diff
@@ -298,64 +315,63 @@ private void saveConfigurationToProject(MavenProject proj, boolean includeDeploy
         }
     }
 
-    private void configurePublic() throws IOException {
-        this.isPublic = prompt.promoteYesNo(false, "Do you want to enable public access for this app?", true);
-    }
-
-    private void configurePublicParent() throws IOException {
-        if (targetProjects.size() == 1) {
-            if (prompt.promoteYesNo(false,
-                    String.format("Do you want to enable public access for this app: %s?",
-                            TextUtils.blue(targetProjects.get(0).getName())), true)) {
-                publicProjects = targetProjects;
-            }
+    private void configurePublic() throws IOException, NoResourcesAvailableException, ExpressionEvaluationException, InvalidConfigurationException {
+        if (this.parentMode) {
+            publicProjects = this.wrapper.handleMultipleCase("configure-public-list", targetProjects, MavenProject::getName);
         } else {
-            publicProjects = prompt.promoteMultipleEntities("project", "Select public accessible apps",
-                    targetProjects, t -> t.getName(), true, " to select NONE",
-                    new ArrayList<MavenProject>());
+            this.appSettings.setIsPublic(this.wrapper.handle("configure-public", false));
         }
+
     }
 
-    private void configureAppName() throws IOException {
-        appName = prompt.promoteString("app name", "${project.artifactId}", "[A-Za-z0-9_\\.]+", true);
+    private void configureAppName() throws IOException, ExpressionEvaluationException, InvalidConfigurationException {
+        if (StringUtils.isNotBlank(appName)) {
```

Comment:
Is there any cases that config will run with `appName` not blank?

Reply:
-DappName